
Author: rmh
Description: _NOT_ against qemu.  This patch is against the FreeBSD patchset
  in their ports collection.  After appliing, some parts of the patchset that
  are FreeBSD-specific need to be removed (yes, read the files and review them).
  Then apply the new patchset to qemu.
  .
  This patch is against a snapshot (20050911) that applies to qemu 0.7.2.
Status: sent to maintainer of FreeBSD qemu port.

diff -ur files.old/patch-bk files/patch-bk
--- files.old/patch-bk	2004-10-25 16:57:30.000000000 +0200
+++ files/patch-bk	2005-09-11 13:31:11.000000000 +0200
@@ -178,8 +178,8 @@
 RCS file: /cvsroot/qemu/qemu/slirp/bootp.h,v
 retrieving revision 1.1
 diff -u -r1.1 bootp.h
---- slirp/bootp.h	22 Apr 2004 00:10:47 -0000	1.1
-+++ slirp/bootp.h	5 Jun 2004 19:34:22 -0000
+--- qemu/slirp/bootp.h	22 Apr 2004 00:10:47 -0000	1.1
++++ qemu/slirp/bootp.h	5 Jun 2004 19:34:22 -0000
 @@ -71,6 +71,7 @@
  #define DHCPOFFER		2
  #define DHCPREQUEST		3
diff -ur files.old/patch-bt files/patch-bt
--- files.old/patch-bt	2004-07-11 13:41:06.000000000 +0200
+++ files/patch-bt	2005-09-11 12:54:47.000000000 +0200
@@ -3,7 +3,7 @@
  #ifndef __APPLE__
  #include <libutil.h>
  #endif
-+#ifdef __FreeBSD__
++#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 +#include <sys/module.h>
 +#endif
  #else
@@ -13,7 +13,7 @@
  
  #endif /* CONFIG_SLIRP */
  
-+#ifdef __FreeBSD__
++#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 +#define LOAD_QUIETLY	1
 +#define LOAD_VERBOSLY	2
 +
@@ -48,7 +48,7 @@
      char *dev;
      struct stat s;
  
-+#ifdef __FreeBSD__
++#if defined(__FreeBSD__) || defined(__FreeBSD_kernel__)
 +    int i, kldtried = 0, enoentcount = 0, err = 0;
 +    char dname[100];
 +#ifdef USE_DEVTAP
diff -ur files.old/patch-fbsd files/patch-fbsd
--- files.old/patch-fbsd	2005-09-10 19:04:42.000000000 +0200
+++ files/patch-fbsd	2005-09-11 14:45:34.000000000 +0200
@@ -85,7 +85,7 @@
  #endif
  
 -#if defined(_BSD)
-+#if defined(_BSD) && !defined(__FreeBSD__)
++#if defined(_BSD) && !defined(__FreeBSD__) && !defined(__FreeBSD_kernel__)
  #define lrint(d)		((int32_t)rint(d))
  #define llrint(d)		((int64_t)rint(d))
  #endif
@@ -94,7 +94,7 @@
  /* Native implementation of soft float functions */
  #include <math.h>
 -#if defined(_BSD) && !defined(__APPLE__)
-+#if defined(_BSD) && !defined(__APPLE__) && \
++#if defined(_BSD) && !defined(__APPLE__) && !defined(__GLIBC__) && \
 +    (!defined(__FreeBSD__) || __FreeBSD_version < 500000)
  #include <ieeefp.h>
 +#if defined(__FreeBSD__)
@@ -114,7 +114,7 @@
  | Software IEC/IEEE floating-point rounding mode.
  *----------------------------------------------------------------------------*/
 -#if defined(_BSD) && !defined(__APPLE__)
-+#if defined(_BSD) && !defined(__APPLE__) && \
++#if defined(_BSD) && !defined(__APPLE__) && !defined(__GLIBC__) && \
 +    (!defined(__FreeBSD__) || __FreeBSD_version < 500000)
  enum {
      float_round_nearest_even = FP_RN,
@@ -134,7 +134,7 @@
  /* native float support */
 -#if (defined(__i386__) || defined(__x86_64__)) && !defined(_BSD)
 +#if (defined(__i386__) || defined(__x86_64__)) && \
-+    (!defined(_BSD) || defined(__FreeBSD__))
++    (!defined(_BSD) || defined(__FreeBSD__) || defined(__GLIBC__))
  #define FLOATX80
  #endif
  #endif /* !CONFIG_SOFTFLOAT */
diff -ur files.old/patch-osdep.c files/patch-osdep.c
--- files.old/patch-osdep.c	2005-05-05 14:41:10.000000000 +0200
+++ files/patch-osdep.c	2005-09-11 14:49:07.000000000 +0200
@@ -3,7 +3,7 @@
  
  #elif defined(USE_KQEMU)
  
-+#ifndef __FreeBSD__
++#ifdef __linux__
  #include <sys/vfs.h>
 +#endif
  #include <sys/mman.h>
@@ -13,7 +13,7 @@
      const char *tmpdir;
      char phys_ram_file[1024];
      void *ptr;
-+#ifndef __FreeBSD__
++#ifdef __linux__
      struct statfs stfs;
  
      if (phys_ram_fd < 0) {
@@ -23,7 +23,7 @@
      }
 +#endif
      size = (size + 4095) & ~4095;
-+#ifndef __FreeBSD__
++#ifdef __linux__
      ftruncate(phys_ram_fd, phys_ram_size + size);
      ptr = mmap(NULL, 
                 size, 
diff -ur files.old/patch-vl.c files/patch-vl.c
--- files.old/patch-vl.c	2005-09-10 19:04:42.000000000 +0200
+++ files/patch-vl.c	2005-09-11 13:50:42.000000000 +0200
@@ -15,7 +15,7 @@
  }
  
 -#if defined(__linux__)
-+#if defined(__linux__) || defined(__FreeBSD__)
++#if defined(__linux__) || defined(__FreeBSD__) || defined(__GLIBC__)
  CharDriverState *qemu_chr_open_pty(void)
  {
      char slave_name[1024];
